### PR TITLE
Modal Redesign

### PR DESCRIPTION
**Pull request by flickz**
This PR redesigns our current modal, creates a new `Modal` component in `StyledModal.js`, the modal uses [react compound pattern](https://medium.com/@Dane_s/react-js-compound-components-a6e54b5c9992) to enable flexibility. 
Both `EditCollectiveDelete` and `EditCollectiveArchive` components have been updated to use the new modal, it also included minor changes in their respective tests.

The old modal file has not been removed as [`TicketConfirmed`](https://github.com/opencollective/opencollective-frontend/blob/master/src/components/TicketsConfirmed.js) still relies on it. (Should be updated later in a new PR).

Below are screenshots of the new modal:

Screenshot from 2019-05-07 16-54-50
Screenshot from 2019-05-07 16-54-56

Ref [#1854](https://github.com/opencollective/opencollective/issues/1854)

### CODE REVIEW

**Comment by znarf** [comment]
Looks good to me. Waiting on @Betree's review.

File: src/components/StyledButton.js [hunk 1]
```diff
@@ -67,6 +68,7 @@ const StyledButtonContent = styled(tag.button)`
   ${textAlign}
   ${width}
   ${height}
+  ${margin}
```
Comment (Betree) [inline src/components/StyledButton.js]: This is not required as we already add `space` which provides a margin prop with `m`, `mx`...etc

See https://styled-system.com/api/#space

File: src/components/StyledModal.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+);
+
+export const ModalFooter = ({ children }) => (
+  <Container>
+    <Divider />
+    {children}
+  </Container>
+);
+
+const Modal = ({ children, show, width, height, onClose }) => {
```
Comment (Betree) [inline src/components/StyledModal.js]: Having `propTypes` for this one will be nice, there's no way to know without looking at the code that `width` and `height` are expected to be strings.

File: src/components/StyledModal.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      height: ${props.height};
+    `}
+`;
+
+const Divider = styled.div`
+  margin: 3rem 0;
+  width: 100%;
+  height: 1px;
+  background-color: #e1e4e6;
+  background-color: var(--silver-four);
```
Comment (Betree) [inline src/components/StyledModal.js]: I don't think that this is necessary, we're progressively moving out of global styles and as we already define `background-color` with the same value above this line has no effect.

File: src/components/StyledModal.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+      <React.Fragment>
+        <ModalWrapper width={width} height={height}>
+          {React.Children.map(children, child => {
+            if (child.type.displayName === 'Header') {
+              return React.cloneElement(child, { onClose });
+            }
+            return child;
+          })}
+        </ModalWrapper>
+        <ModalOverlay show={show} onClick={onClose} />
```
Comment (Betree) [inline src/components/StyledModal.js]: As we return `null` if `show=false`, there's no need to pass it here nor to have `display: none` in `ModalOverlay`

**Pull request merged**